packages/client/src/performance/performanceDataReducer.ts

```diff
@@ -0,0 +1,173 @@
+  variables: Record<string, string>,
+  query: DocumentNode
+) {
+  const stored = await storage.getItem(PERFORMANCE_DATA)
+  const key = createKey({ operationName, variables })
+  return {
+    data: stored ? JSON.parse(stored)[key] : null,
+    operationName,
+    variables,
+    query
+  }
+}
+
+async function updateStoredData(
+  data: any,
+  operationName: string,
+  variables: Record<string, string>
+) {
+  const stored = await storage.getItem(PERFORMANCE_DATA)
+  const json = stored ? JSON.parse(stored) : {}
+  const key = createKey({ operationName, variables })
+  const newJson = { ...json, [key]: data?.data }
+  await storage.setItem(PERFORMANCE_DATA, JSON.stringify(newJson))
+}
+
+const prefetchLocationStatisticsCmd = (userDetails: IUserDetails) => {
+  const locationId = getDefaultPerformanceLocationId(userDetails)
+  let variables: Record<string, any> = {
+    populationYear: new Date().getFullYear(),
+    event: 'birth',
+    status: [
+      'IN_PROGRESS',
+      'DECLARED',
+      'REJECTED',
+      'VALIDATED',
+      'WAITING_VALIDATION',
+      'REGISTERED'
+    ],
+    officeSelected: false
+  }
+  if (locationId) {
+    variables = {
+      locationId,
+      ...variables
+    }
+  }
+  return Cmd.action(
+    isAvailable('getLocationStatistics', variables, PERFORMANCE_STATS)
+  )
+}
+
+export const performanceDataReducer = (
+  state: PerformanceDataState = {},
+  action: Action | ProfileAction
+):
+  | PerformanceDataState
+  | Loop<PerformanceDataState, Action | ProfileAction> => {
+  switch (action.type) {
+    case 'PROFILE/USER_DETAILS_AVAILABLE': {
+      return loop(
+        state,
+        Cmd.list([prefetchLocationStatisticsCmd(action.payload)])
+      )
+    }
+    case 'PERFORMANCE/QUERY_DATA_AVAILABLE': {
+      const { operationName, variables, query } = action.payload
+      return loop(
+        state,
+        Cmd.run(getStoredData, {
+          successActionCreator: read,
+          args: [operationName, variables, query]
+        })
+      )
+    }
+    case 'PERFORMANCE/READ_QUERY_DATA': {
+      const { data, operationName, variables, query } = action.payload
+      const key = createKey({ operationName, variables })
+      return loop(
+        {
+          ...state,
+          [key]: {
+            loading: !data,
+            data,
+            error: null
+          }
+        },
+        Cmd.action(fetch(operationName, variables, query))
+      )
+    }
+    case 'PERFORMANCE/FETCH_QUERY_DATA': {
+      const { operationName, variables, query } = action.payload
+      const key = createKey({ operationName, variables })
+      return loop(
+        {
+          ...state,
+          [key]: {
+            loading: true,
+            data: null,
+            error: null
+          }
+        },
+        Cmd.run(() => client.query({ query, variables }), {
+          successActionCreator: (data) =>
+            fetchSuccess(data, operationName, variables)
+        })
+      )
+    }
```

Comment:
Let's say we have data available for a particular key, so we will update our redux state with that, then call fetch in line 132. What fetch will do is clear the data for that particular key, turn loading into true and refetch the data. We probably don't want this behavior right?

Reply:
Great spot, @Zangetsu101, thanks. I will update the loading value.  I think `!state[key]?.data`, would do the work 🤔

Reply:
yeah it should do the trick